pkg/proxy/apis/config/validation/validation.go

```diff
@@ -163,14 +163,22 @@ func validateKubeProxyConntrackConfiguration(config kubeproxyconfig.KubeProxyCon
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("Min"), config.Min, "must be greater than or equal to 0"))
 	}
 
-	if config.TCPEstablishedTimeout.Duration < 0 {
+	if config.TCPEstablishedTimeout != nil && config.TCPEstablishedTimeout.Duration < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("TCPEstablishedTimeout"), config.TCPEstablishedTimeout, "must be greater than or equal to 0"))
 	}
 
-	if config.TCPCloseWaitTimeout.Duration < 0 {
+	if config.TCPCloseWaitTimeout != nil && config.TCPCloseWaitTimeout.Duration < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("TCPCloseWaitTimeout"), config.TCPCloseWaitTimeout, "must be greater than or equal to 0"))
 	}
 
+	if config.UDPTimeout.Duration < 0 {
```

Comment:
You added nil-checks to the above cases but not these new ones, why?

Reply:
TCPEstablishedTimeout and TCPCloseWaitTimeout are pointer types `*metav1.Duration`.

The new ones - UDPTimeout and UDPStreamTimeout are value types `metav1.Duration` so I didn't add the nil-check here.

Reply:
Why did we opt to do them differently?  I'm not sure the originals needed to be pointers, but half-and-half is probably the worst choice.

Reply:
> but half-and-half is probably the worst choice.

I agree with this, but we already have a mixture of both.  
There [8 fields with](https://grep.app/search?q=%20metav1.Duration%24&regexp=true&filter%5Brepo%5D%5B0%5D=kubernetes/kubernetes&filter%5Bpath%5D%5B0%5D=pkg/proxy/apis/config/&filter%5Blang%5D%5B0%5D=Go) of type `metav1.Duration` and [2 of type](https://grep.app/search?q=%5C%2Ametav1.Duration%24&regexp=true&filter%5Brepo%5D%5B0%5D=kubernetes/kubernetes&filter%5Bpath%5D%5B0%5D=pkg/proxy/apis/config/&filter%5Blang%5D%5B0%5D=Go) `*metav1.Duration`.

Currently we can't use can't use objects of type `*metav1.Duration` directly in pflag without setting any default value.
```go
var param *metav1.Duration

fs.DurationVar(&param.Duration, ...)
                    ^^ this will panic as param is nil
``` 

https://github.com/kubernetes/kubernetes/pull/55261 changed types of  **TCPCloseWaitTimeout**  and **TCPEstablishedTimeout** to pointer types to allow zero values.

Reply:
Ahh, I see.  For TCP, we have default values which we assert.  So we need to know the difference between "uspecified" (meaning "set the default value") and "specified as 0" (meaning "do not change").

We're not proposing default values for these new ones, so 0, whether specified or not, means "do not change it".  Right?

And setting a default at this point is likely to end in tears for someone.

So adding the nil-check is actually wrong - the TCP fields *must* have a value by this point, and allowing nil to slip through is actually bad.

Can you please remove those nil-checks and add comments like "config.TCPEstablishedTimeout has a default value, so can't be nil"?

We should fix the API comment-docs, but we can do that later.

Reply:
> Can you please remove those nil-checks and add comments like "config.TCPEstablishedTimeout has a default value, so can't be nil"?

sure.

Reply:
We went through these in another review, the pointers come from here https://github.com/kubernetes/kubernetes/pull/55261 , but should not be there honestly, are confusing

Reply:
I think the pointer-ness DOES matter here - we need to know the difference between "uspecified" (meaning "set the default value") and "specified as 0" (meaning "do not change"), but these new fields do not have defaults.

Reply:
> I think the pointer-ness DOES matter here - we need to know the difference between "uspecified" (meaning "set the default value") and "specified as 0" (meaning "do not change"), but these new fields do not have defaults.

@thockin we tried to add  a new `cliflag.MetaDuration` flag - https://github.com/kubernetes/kubernetes/pull/120489 for the same purpose and this can be directly consumed by flag avoiding any panics.